Make testing._internal.common_utils safe to import

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99473
* __->__ #99659

In edge cases in CI, SLOW_TESTS_FILE is defined but does not point to an existing file.

Guessing this is due to a test case that opens a subprocses and cwd's but doesn't clean its env.

We shouldn't make importing common_utils fail, so issue a warning and proceed.

wconstab committed:
Make testing._internal.common_utils safe to import

In edge cases in CI, SLOW_TESTS_FILE is defined but does not point to an existing file.

Guessing this is due to a test case that opens a subprocses and cwd's but doesn't clean its env.

We shouldn't make importing common_utils fail, so issue a warning and proceed.

[ghstack-poisoned]

ezyang commented on torch/testing/_internal/common_utils.py:
```diff
@@ -112,13 +112,19 @@
+        with open(filename, 'r') as fp:
+            data = json.load(fp)
+            return data
+    print(f"Attempted to load {filename} but it does not exist.")
```
Use logging instead?

malfet commented on torch/testing/_internal/common_utils.py:
```diff
@@ -112,13 +112,19 @@
+def maybe_load_json(filename):
+    if os.path.exists(filename):
+        with open(filename, 'r') as fp:
+            data = json.load(fp)
+            return data
```
Nit
```suggestion
            return json.load(fp)
```

malfet commented on torch/testing/_internal/common_utils.py:
```diff
@@ -112,13 +112,19 @@
 slow_tests_dict = {}
 
+def maybe_load_json(filename):
+    if os.path.exists(filename):
```
Not sure if we want to raise an exception here if somebody to call this function with "." or ".."
```suggestion
    if os.path.isfile(filename):
```

wconstab committed:
Update on "Make testing._internal.common_utils safe to import"


In edge cases in CI, SLOW_TESTS_FILE is defined but does not point to an existing file.

Guessing this is due to a test case that opens a subprocses and cwd's but doesn't clean its env.

We shouldn't make importing common_utils fail, so issue a warning and proceed.

[ghstack-poisoned]

wconstab commented:
@pytorchbot merge
